crates/runtime/src/dataconnector/github.rs

```diff
@@ -0,0 +1,324 @@
+}
+
+impl GithubTableArgs for CommitTableArgs {
+    fn get_graphql_values(&self) -> (String, String, usize) {
+        let query = format!(
+            r#"{{
+                repository(owner: "{owner}", name: "{name}") {{
+                    defaultBranchRef {{
+                        target {{
+                            ... on Commit {{
+                                history(first: 100) {{
+                                    pageInfo {{
+                                        hasNextPage
+                                        endCursor
+                                    }}
+                                    nodes {{
+                                        message
+                                        messageHeadline
+                                        messageBody
+                                        oid
+                                        additions
+                                        deletions
+                                        id
+                                        committedDate
+                                        authorName: author {{
+                                            name
+                                        }}
+                                        authorEmail: author {{
+                                            email
+                                        }}
+                                    }}
+                                }}
+                            }}
+                        }}
+                    }}
+                }}
+            }}"#,
+            owner = self.owner,
+            name = self.repo
+        );
+        (
+            query,
+            "/data/repository/defaultBranchRef/target/history/nodes".to_string(),
+            1,
+        )
+    }
+}
+
+// TODO: implement PR filters from https://docs.github.com/en/graphql/reference/objects#repository `Arguments for issues`
+pub struct IssueTableArgs {
+    pub owner: String,
+    pub repo: String,
+}
+
+impl GithubTableArgs for IssueTableArgs {
+    fn get_graphql_values(&self) -> (String, String, usize) {
+        let query = format!(
+            r#"{{
+                repository(owner: "{owner}", name: "{name}") {{
+                    issues(first: 100) {{
+                        pageInfo {{
+                            hasNextPage
+                            endCursor
+                        }}
+                        nodes {{
+                            id
+                            title
+                            url
+                            author: author {{ login }}
+                            body
+                            number
+                            createdAt
+                            updatedAt
+                            closedAt
+                            state
+                            milestone {{ milestone_id: id, milestone_title: title }}
+                            labels(first: 100) {{ labels: nodes {{ name }} }}
```

Comment:
Wondering if we can flatten labels from a list of structs to a list of Utf8. This should simplify working with them. (if there is only name/utf8 field)

Reply:
As above.